eng/pipelines/publish-pipeline-artifact-shim.yml

```diff
@@ -0,0 +1,25 @@
+parameters:
+  inputs: {}
+  displayName: 'Publish Pipeline Artifact'
+  condition: succeeded()
+  continueOnError: true
+
+steps:
+- ${{ if ne(variables['System.TeamProject'], 'public') }}:
+  # If we are using this in a project other than the public one, use the 1es task.
+  - task: 1es.PublishPipelineArtifact@1
+    inputs:
+      targetPath: ${{ parameters.inputs.targetPath }}
+      artifactName: ${{ parameters.inputs.artifactName }}
```

Comment:
Juan mentioned we can append the attempt iteration to the name, so maybe here and L22 we can add `_Attempt$(System.JobAttempt)`

`${{ parameters.inputs.artifactName }}_Attempt$(System.JobAttempt)` maybe?

Reply:
This change will force other changes below where we download the artifacts in L183, L188.  I'm going to think about this a bit.

Reply:
I thought these publish steps were after, as publish-pipeline-artifact-shim.yml is used in L237, L258, L276 in build.yml

Reply:
It is used to upload the artifacts (Build_ suffix), the logs (Logs_) and DiagnosticsArtifacts_ (not sure they are).

The testOnly legs download the Build_xxx artifacts from the previous build leg. Adding the retry attempt will work if it is added to the download step at L179.

Reply:
Adding it to the Log_ publishing should be enough.